platform/node/src/node_map.cpp

```diff
@@ -539,18 +539,14 @@ void NodeMap::renderFinished() {
 
         request->runInAsyncScope(target, callback, 1, argv);
     } else if (img.data) {
+        auto image_size = img.bytes();
         v8::Local<v8::Object> pixels = Nan::NewBuffer(
-            reinterpret_cast<char *>(img.data.get()), img.bytes(),
-            // Retain the data until the buffer is deleted.
-            [](char *, void * hint) {
-                delete [] reinterpret_cast<uint8_t*>(hint);
-            },
-            img.data.get()
-        ).ToLocalChecked();
-        if (!pixels.IsEmpty()) {
-            img.data.release();
-        }
-
+                                           reinterpret_cast<char*>(img.data.release()),
+                                           image_size,
+                                           [](char* buf, void*) { delete[] buf; },
+                                           nullptr)
+                                           .ToLocalChecked();
+        Nan::AdjustExternalMemory(image_size);
```

Comment:
Questions I have:

 - This adds `Nan::AdjustExternalMemory(image_size);` but I think we will also need a paired `Nan::AdjustExternalMemory(-image_size);` inside the `[](char* buf, void*)` deleter callback. Otherwise v8 will not know the memory has been releases and may attempt to garbage collect too often, hurting performance.
 - This also changed from `[](char *, void * hint) {` to `[](char* buf, void*) {`. Why that change?
 - This also removed the `if (!pixels.IsEmpty()) {` check which @flippmoke recently added. Why remove that?

Reply:
> Questions I have:
> 
>     * This adds `Nan::AdjustExternalMemory(image_size);` but I think we will also need a paired `Nan::AdjustExternalMemory(-image_size);` inside the `[](char* buf, void*)` deleter callback. Otherwise v8 will not know the memory has been released and may attempt to garbage collect too often, hurting performance.

Hmm.. I'd be surprised if we need to do that, but good call! We should check implementation, definitely.
> 
>     * This also changed from `[](char *, void * hint) {` to `[](char* buf, void*) {`. Why that change?

First argument is `char*` pointer to the memory we passed in the ctor and second is  `void*` is anything extra that is needed for clean-up. I don't see why passing same pointer again via `hint` is needed. I guess this was just copy/pasted from somewhere  else. In `node_mapnik` passing `std::string*` is required for proper cleanup (calling std::string dtor). 


> 
>     * This also removed the `if (!pixels.IsEmpty()) {` check which @flippmoke recently added. Why remove that?
Why do we need this check? @flippmoke mentioned on slack  "If the result is empty is might not have taken ownership of the ptr" but I don't think this is the case. The only condition where ownership is not transferred is if `Napi::Buffer::New()` throws an exception. 
@flippmoke - do you have more insights, maybe I'm missing something?